lib/fastlane/plugin/revenuecat_internal/actions/insert_changelog_of_older_version_action.rb

```diff
@@ -0,0 +1,125 @@
+require 'fastlane/action'
+require 'fastlane_core/configuration/config_item'
+require_relative '../helper/revenuecat_internal_helper'
+
+module Fastlane
+  module Actions
+    class InsertChangelogOfOlderVersionAction < Action
+      def self.run(params)
+        sdk_version = params[:sdk_version]
+        changelog_path = params[:changelog_path]
+        changelog_latest_path = params[:changelog_latest_path]
+        repo_name = params[:repo_name]
+        base_branch = params[:base_branch]
+        github_pr_token = params[:github_pr_token]
+        hybrid_common_version = params[:hybrid_common_version]
+        append_phc_version = params[:append_phc_version]
+        dry_run = params[:dry_run]
+
+        if Helper::RevenuecatInternalHelper.older_than_latest_published_version?(sdk_version)
+          UI.important("Version #{sdk_version} is older than the latest published version. Proceeding with changelog insertion into #{base_branch}.")
+
+          current_branch = Actions.git_branch
+
+          changelog_content = File.read(changelog_latest_path)
+
+          if dry_run
+            UI.important("Dry run mode enabled. No changes will be made.")
+          end
+
+          Helper::RevenuecatInternalHelper.create_or_checkout_branch(base_branch)
+
+          changelog_update_branch_name = "changelog/#{sdk_version}"
+
+          unless dry_run
+            Helper::RevenuecatInternalHelper.create_new_branch_and_checkout(changelog_update_branch_name)
+          end
+
+          final_version_number = Helper::VersioningHelper.append_phc_version_if_necessary(
+            append_phc_version,
+            false,
+            hybrid_common_version,
+            sdk_version
+          )
+
+          Helper::RevenuecatInternalHelper.insert_old_version_changelog_in_main(final_version_number, changelog_content, changelog_path)
+
+          if dry_run
+            updated_changelog = File.read(changelog_path)
+            UI.important("The updated changelog would look like this:\n#{updated_changelog}")
```

Comment:
Hmm won't this print the whole changelog, which can be pretty huge? I'm not sure if it's necessary or whether we should just print what version was added...

Reply:
Right... Maybe it's ok then... We could maybe print the result of `git diff`... but this is probably ok for now 👍